### PR TITLE
Login: Corrections, session added, multilanguage supported

### PR DESCRIPTION
**Pull request by adCBueno**

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -31,4 +31,7 @@
     <string name="name_app">RoutesApp</string>
     <string name="tv_welcome">Welcome back!</string>
     <string name="btn_sign_outd">Sign Out</string>
+    <string name="login_email_empty">Email with invalid format</string>
+    <string name="login_password_empty">Empty password</string>
+    <string name="login_success">Login Successfully</string>
```
Comment (YawarValeriano) [inline app/src/main/res/values/strings.xml:36]: This should match the translation from values-es

File: app/src/test/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModelTest.kt [hunk 2]
```diff
@@ -48,6 +49,68 @@ class LoginViewModelTest : TestCase() {
         viewModel = LoginViewModel(fakeManager)
     }
 
+    @Test
+    fun `Given an empty or invalid format of email`() {
```
Comment (YawarValeriano) [inline app/src/test/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModelTest.kt]: We need to provide what is going to be te expected result in the description

File: app/src/test/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModelTest.kt [hunk 3]
```diff
@@ -48,6 +49,68 @@ class LoginViewModelTest : TestCase() {
...
+        runBlocking {
+            viewModel.validateFields(email, password)
+            val result = fakeManager.signInWithEmailAndPassword(email, password)
+            val resultTwo = fakeManager.signInWithEmailAndPassword(emailTwo, passwordTwo)
+            assertNotSame(result,resultTwo)
+            val observer = Observer<Boolean> {}
+            try {
+                viewModel.loginIsSuccessful.observeForever(observer)
+                val value = viewModel.loginIsSuccessful.getOrAwaitValue()
+                assertTrue(value)
```
Comment (YawarValeriano) [inline app/src/test/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModelTest.kt]: If the test is given a invalid email it should fail in loggin

File: app/src/test/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModelTest.kt [hunk 4]
```diff
@@ -48,6 +49,68 @@ class LoginViewModelTest : TestCase() {
...
+            } finally {
+                viewModel.loginIsSuccessful.removeObserver(observer)
+            }
+        }
+    }
+
+    @Test
+    fun `Given an empty password`() {
```
Comment (YawarValeriano) [inline app/src/test/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModelTest.kt]: Same here with the description

File: app/src/test/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModelTest.kt [hunk 5]
```diff
@@ -48,6 +49,68 @@ class LoginViewModelTest : TestCase() {
...
+        runBlocking {
+            viewModel.validateFields(email, password)
+            val result = fakeManager.signInWithEmailAndPassword(email, password)
+            val resultTwo = fakeManager.signInWithEmailAndPassword(emailTwo, passwordTwo)
+            assertNotSame(result,resultTwo)
+            val observer = Observer<Boolean> {}
+            try {
+                viewModel.loginIsSuccessful.observeForever(observer)
+                val value = viewModel.loginIsSuccessful.getOrAwaitValue()
+                assertTrue(value)
```
Comment (YawarValeriano) [inline app/src/test/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModelTest.kt]: Same her about succes with wrong credentials

File: app/src/test/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModelTest.kt [hunk 6]
```diff
@@ -48,6 +49,68 @@ class LoginViewModelTest : TestCase() {
...
+            } finally {
+                viewModel.loginIsSuccessful.removeObserver(observer)
+            }
+        }
+    }
+
+    @Test
+    fun `Given a correct email and password`() {
```
Comment (YawarValeriano) [inline app/src/test/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModelTest.kt]: Same here about description

**Review by YawarValeriano — CHANGES_REQUESTED** [review]
Also check lintin error, because bitrise fails

File: app/src/test/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModelTest.kt (hunk 2 above)
Comment (adCBueno) [reply app/src/test/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModelTest.kt]: Corrected :)

File: app/src/test/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModelTest.kt (hunk 3 above)
Comment (adCBueno) [reply app/src/test/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModelTest.kt]: Corrected

File: app/src/test/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModelTest.kt (hunk 4 above)
Comment (adCBueno) [reply app/src/test/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModelTest.kt]: Added

File: app/src/test/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModelTest.kt (hunk 5 above)
Comment (adCBueno) [reply app/src/test/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModelTest.kt]: Corrected

File: app/src/test/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModelTest.kt (hunk 6 above)
Comment (adCBueno) [reply app/src/test/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModelTest.kt]: Added

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (adCBueno) [reply app/src/main/res/values/strings.xml:36]: Corrected

**Review by jmacboy — APPROVED** [review]

**Review by YawarValeriano — APPROVED** [review]

**Pull request merged**